### ISSUE TITLE
Linking to a media file in content studio in preview issue

### ISSUE THREAD

**Issue by pfrang** [enonic/nextjs-adapter#172]
A url like https://sio-xp7test-master.enonic.cloud/api/_/attachment/inline/8824ac57-dde4-4427-9dbe-203561cf28d8:826d1523bf17a1d90474f2e36d44336e2432c26d/Postlister%202023.pdf is not able to be navigated to in content studio with preview

Testing locally, when hovering inside CS I see that the url is the correct one, but when clicking its immedeately returning cant connect (picture 2)

![image](https://user-images.githubusercontent.com/85552255/234041006-c38adec3-616c-45b8-821e-f498c83ac0f4.png)

![image](https://user-images.githubusercontent.com/85552255/234041212-6774b839-89da-4c4c-8f53-7c4ec2537332.png)

**Comment by sigdestad** [comment]
Hmm.. Don't think you are supposed to wrap links that are generated by XP in getUrl. enonicAdapter handles these automatically afaik? This function is only needed for your custom links, i.e. static assets or custom page links?

@pmi - any comments?

**Comment by sigdestad** [comment]
And you only need to wrap in getUrl if it is important for it to work inside of CS.

**Comment by sigdestad** [comment]
What do you mean "pointing to another content type"?

**Comment by sigdestad** [comment]
Seems I'm mistaken here - so getUrl should be used for all links apparently. However, there is something very wrong with the link you generate when seen from Admin.

And, to answer your original problem. You will never be able to "navigate" to an item in content studio by generating links to downloading a file.

**Comment by sigdestad** [comment]
What you can probably do if you want clicking these links in CS navigate to the media content, is have a check that says "if preview" create URL to content, rather than the attachment download. Notice, then editor will never be able to actually download the file - which is not what you really want IMHO.

**Comment by pmi** [comment]
I agree with Thomas here that the `/_/attachment/` link should download the file, not preview it.
You can create a valid download link by adding a Text component and inserting a download link with an insert link dialog.
